train_gpt2.cu

```diff
@@ -2985,6 +2990,10 @@ void load_state(int* step, GPT2* model, DataLoader* loader, const char* filename
     cudaCheck(cudaMemcpy(model->m_memory, cpu_buffer, shard_num_parameters * sizeof(float), cudaMemcpyHostToDevice));
     freadCheck(cpu_buffer, sizeof(float), shard_num_parameters, state_file);
     cudaCheck(cudaMemcpy(model->v_memory, cpu_buffer, shard_num_parameters * sizeof(float), cudaMemcpyHostToDevice));
+    if (state_header[4] == 1) {  // if we used master weights during the state save
```

Comment:
I think this might crash, because we might not have allocated the corresponding memory buffer yet

Reply:
nice catch, fixed!